src/core/server/saved_objects/migrationsv2/integration_tests/rewriting_id.test.ts

```diff
@@ -0,0 +1,240 @@
+    index,
+    body: {
+      query: {
+        bool: {
+          should: [
+            {
+              term: { type: 'foo' },
+            },
+            {
+              term: { type: 'bar' },
+            },
+            {
+              term: { type: 'legacy-url-alias' },
+            },
+          ],
+        },
+      },
+    },
+  });
+
+  return body.hits.hits
+    .map((h) => ({
+      ...h._source,
+      id: h._id,
+    }))
+    .sort(sortByTypeAndId);
+}
+
+function createRoot() {
+  return kbnTestServer.createRootWithCorePlugins(
+    {
+      migrations: {
+        skip: false,
+        enableV2: true,
+      },
+      logging: {
+        appenders: {
+          file: {
+            type: 'file',
+            fileName: logFilePath,
+            layout: {
+              type: 'json',
+            },
+          },
+        },
+        loggers: [
+          {
+            name: 'root',
+            appenders: ['file'],
+          },
+        ],
+      },
+    },
+    {
+      oss: true,
+    }
+  );
+}
+
+describe('migration v2', () => {
+  let esServer: kbnTestServer.TestElasticsearchUtils;
+  let root: Root;
+
+  beforeAll(async () => {
+    await removeLogFile();
+  });
+
+  afterAll(async () => {
+    if (root) {
+      await root.shutdown();
+    }
+    if (esServer) {
+      await esServer.stop();
+    }
+
+    await new Promise((resolve) => setTimeout(resolve, 10000));
+  });
+
+  it('rewrites id deterministically for SO with namespaceType: "multiple" and "multiple-isolated"', async () => {
+    const migratedIndex = `.kibana_${pkg.version}_001`;
+    const { startES } = kbnTestServer.createTestServers({
+      adjustTimeout: (t: number) => jest.setTimeout(t),
+      settings: {
+        es: {
+          license: 'trial',
+          // original SO:
+          // [
+          //   { id: 'foo:1', type: 'foo', foo: { name: 'Foo 1 default' } },
+          //   { id: 'spacex:foo:1', type: 'foo', foo: { name: 'Foo 1 spacex' }, namespace: 'spacex' },
+          //   {
+          //     id: 'bar:1',
+          //     type: 'bar',
+          //     bar: { nomnom: 1 },
+          //     references: [{ type: 'foo', id: '1', name: 'Foo 1 default' }],
+          //   },
+          //   {
+          //     id: 'spacex:bar:1',
+          //     type: 'bar',
+          //     bar: { nomnom: 2 },
+          //     references: [{ type: 'foo', id: '1', name: 'Foo 1 spacex' }],
+          //     namespace: 'spacex',
+          //   },
+          // ];
+          dataArchive: Path.join(__dirname, 'archives', '7.13.0_so_with_multiple_namespaces.zip'),
+        },
+      },
+    });
+
+    root = createRoot();
+
+    esServer = await startES();
+    const coreSetup = await root.setup();
+
+    coreSetup.savedObjects.registerType({
+      name: 'foo',
+      hidden: false,
+      mappings: { properties: { name: { type: 'text' } } },
+      namespaceType: 'multiple',
+      convertToMultiNamespaceTypeVersion: '8.0.0',
```

Comment:
I think the test will fail when backporting to `7.x` as the type will not be converted given that `8.0.0` > `7.13.0`

Reply:
ok, I will change it to `7.13.0`, but then we will have to update the default https://github.com/elastic/kibana/blob/dc475c9cf2a7f42c6b7df921a6dc8613765c80d7/src/core/server/saved_objects/migrations/core/document_migrator.ts#L70
@jportner FYI

Reply:
Thanks for the ping. We **don't** want to change `DEFAULT_MINIMUM_CONVERT_VERSION`; that's specifically in set to ensure we _don't_ allow consumers to convert any existing objects before 8.0, because conversion (regenerating object IDs) is considered to be a major breaking change. This is one thing that @rudolf asked me to add during my initial implementation.

This constraint is configurable for integration tests by using the `minimumConvertVersion` parameter when creating the DocumentMigrator, instead of changing this deafult, can you update `rewriting_id.test.ts` to use `minimumConvertVersion`?

Reply:
AFAIK we can't, as it's an integration and not a unit test.

Not sure which option remains except skipping the test on branches lower than master.

Reply:
Ah, I was using this in an integration test (`test/api_integration/apis/saved_objects/migrations.ts`), but that test was creating a `DocumentMigrator` and `IndexMigrator` instance directly. Looking at it again, `rewriting_id.test.ts` goes about this differently and it would be difficult to change without exposing this at runtime. I think skipping the test in the 7.x branch as mentioned in https://github.com/elastic/kibana/pull/97222#issuecomment-826572126 is fine, though.